repose-aggregator/components/filters/body-patcher-filter/src/main/scala/org/openrepose/filters/bodypatcher/BodyPatcherFilter.scala

```diff
@@ -0,0 +1,150 @@
+    val responsePatches: List[Patch] = filterResponseChanges(pathChanges)
+
+    //patch the request
+    val patchedRequest: HttpServletRequest = determineContentType(httpRequest.getContentType) match {
+      case Json =>
+        val jsonPatches: List[String] = filterJsonPatches(requestPatches)
+        if (jsonPatches.nonEmpty) {
+          logger.debug("Applying json patches on the request")
+          val patchedValue: JsValue = applyJsonPatches(PJson.parse(httpRequest.getInputStream), jsonPatches)
+          new HttpServletRequestWrapper(httpRequest, new ServletInputStreamWrapper(new ByteArrayInputStream(PJson.stringify(patchedValue).getBytes)))
+        } else {
+          httpRequest
+        }
+      case Xml =>
+        //todo: implement xml support
+        httpRequest
+      case Other =>
+        httpRequest
+    }
+
+
+    //check if we might do work to the response
+    if (responsePatches.isEmpty) {
+      chain.doFilter(patchedRequest, httpResponse)
+    } else {
+      val wrappedResponse: HttpServletResponseWrapper = new HttpServletResponseWrapper(httpResponse, PASSTHROUGH, MUTABLE)
+      chain.doFilter(patchedRequest, wrappedResponse)
+      wrappedResponse.uncommit()
+
+      determineContentType(wrappedResponse.getContentType) match {
+        case Json =>
+          val jsonPatches: List[String] = filterJsonPatches(responsePatches)
+          if (jsonPatches.nonEmpty) {
+            logger.debug("Applying json patches on the response")
+
+            val contentStream = wrappedResponse.getOutputStreamAsInputStream
+            val contentValue = if (contentStream.available() > 0) {
+              PJson.parse(contentStream)
+            } else {
+              PJson.obj()
+            }
+            val patchedValue: JsValue = applyJsonPatches(contentValue, jsonPatches)
+            wrappedResponse.setOutput(new ByteArrayInputStream(PJson.stringify(patchedValue).getBytes))
+          }
+        case Xml => //todo implement xml support
+        case Other =>
+      }
+      wrappedResponse.commitToResponse()
+    }
+  }
+
+  def filterChanges(request: HttpServletRequest): List[ChangeDetails] = {
+    val urlPath: String = new URL(request.getRequestURL.toString).getPath
+    configuration.getChange.asScala.toList
+        .filter(_.getPath.r.findFirstIn(urlPath).isDefined)
+  }
+
+  def filterRequestChanges(changes: List[ChangeDetails]): List[Patch] = {
+    changes.flatMap(change => Option(change.getRequest))
+  }
+
+  def filterResponseChanges(changes: List[ChangeDetails]): List[Patch] = {
+    changes.flatMap(change => Option(change.getResponse))
+  }
+
+  def determineContentType(contentType: String): ContentType = {
+    //magic code from stack overflow, i'm a terrible person, http://stackoverflow.com/questions/4636610/how-to-pattern-match-using-regular-expression-in-scala
+    implicit class Regex(sc: StringContext) {
+      def r = new util.matching.Regex(sc.parts.mkString, sc.parts.tail.map(_ => "x"): _*)
+    }
```

Comment:
I thought Scala's regex got much better in 2.11.  Is this magic still necessary?